apps/documentation/src/components/MdxProvider/MdxTitle/MdxTitle.tsx

```diff
@@ -14,28 +14,22 @@ export default function MdxTitle(props: TitleProps): JSX.Element {
 }
 
 function idGen(children: ReactNode): string {
-  if (Array.isArray(children)) {
-    const result = children
-      .map((child) => {
-        if (typeof child === 'string') {
-          return child
-        }
-        if (typeof child === 'object' && child !== null && 'props' in child) {
-          const childWithProps = child as { props?: { children?: ReactNode } }
-          return typeof childWithProps.props?.children === 'string'
-            ? childWithProps.props.children
-            : ''
-        }
-        return ''
-      })
-      .join('')
+  const getTextContent = (node: ReactNode): string => {
+    if (typeof node === 'string') return node
+    if (typeof node === 'object' && node !== null && 'props' in node) {
+      const child = node as { props?: { children?: ReactNode } }
+      return getTextContent(child.props?.children)
+    }
+    return ''
+  }
 
+  if (Array.isArray(children)) {
```

Comment:
You can remove the if and use a ternary here to dedupe the `toLowerCase` and `replace` code:

```ts
const textContent = Array.isArray(children)
    ? children.map(getTextContent).join('')
    : getTextContent(children)

return textContent.toLowerCase().replace(/\s+/g, '-')
```